client/src/pages/SeriesDetailsPage.js

```diff
@@ -2,24 +2,33 @@ import { useEffect, useState } from 'react'
 import { useParams } from 'react-router'
 import { Link } from 'react-router-dom'
 import styled from 'styled-components/macro'
+import { ReactComponent as IconArrowLeft } from '../assets/icons/long-arrow-alt-left-solid.svg'
+import ButtonWatchlist from '../components/ButtonWatchlist'
 import Poster from '../components/Poster'
-import IconLeftArrow from '../components/ui/IconLeftArrow'
 
-export default function SeriesDetails() {
+export default function SeriesDetails({
+  series,
+  handleWatchlist,
+  handleNewSeries,
+}) {
   const { id } = useParams()
   const [seriesDetails, setSeriesDetails] = useState([])
   const [cast, setCast] = useState([])
 
   useEffect(() => {
-    fetch(`/api/series/${id}`)
-      .then(res => res.json())
-      .then(data => {
-        setSeriesDetails(data)
-      })
-      .catch(error => {
-        console.error('Error:', error)
-      })
-  }, [id])
+    const filteredItem = series.filter(el => el.id === Number(id))
+    filteredItem.length !== 0
+      ? setSeriesDetails(filteredItem[0])
```

Comment:
You could use the `find()` instead of `filter()` as it returns the object you are looking for and not an array. That way you'd save yourself iterating over the `filteredItem` in line 21.